pandas/core/sparse/frame.py

```diff
@@ -846,6 +846,13 @@ def applymap(self, func):
         return self.apply(lambda x: lmap(func, x))
 
 
+def _nan_to_np_nan(value):
+    """Normalize nan values to singleton np.NaN object so that when NaNs are
```

Comment:
Because one of the tests apparently sets a `float('nan')` (`is not np.nan`) as a valid column name, and `{float('nan'): 'foo'}[np.nan]` is a KeyError. I didn't know what else to do. DataFrame apparently handles all cases of nan specially; it seemed easiest to force singleton for dict.